src/components/TreeView.js

```diff
@@ -4,26 +4,40 @@
 import React from 'react';
 import _ from 'lodash';
 import {EasySelect} from './EasySelect';
+import SIGNATURE from '../constants/signature';
 
 // @param {array} props.nodes - Array of TreeView compatible nodes
 export default class TreeView extends React.Component {
   render() {
-    let {nodes, className} = this.props;
+    let {nodes, className, fetchedSigners, parent} = this.props;
     let rootClass = 'TreeView ' + (className) ? className : '';
 
+
     let result = <div className={rootClass}>
       {_.map(Array.prototype.slice.call(nodes), (node, index) => {
         let childNodes;
 
+        let sep = '.';
+        if (!parent || node.type.charAt(0) == '[') {
+          sep = '';
+        }
+ 
+        let position = (parent ? parent + sep : '') + node.type;
+
+        if (position == 'TransactionEnvelope.signatures') {
+          checkSignatures(nodes, fetchedSigners);
```

Comment:
Use `node` here so that you don't need to use `nodes[1]` inside `checkSignatures`..